apps/hannk/halide/CMakeLists.txt

```diff
@@ -8,49 +8,49 @@ set(CMAKE_CXX_EXTENSIONS NO)
 
 # ----------------------------
 
-if (NOT CMAKE_CROSSCOMPILING)
+add_custom_target(hannk_tools)
+
+function(_export_hannk_tool TARGET)
+    add_dependencies(hannk_tools ${TARGET})
+    export(TARGETS ${TARGET}
+           NAMESPACE hannk::tools::
+           APPEND FILE "${hannk_BINARY_DIR}/hannk_tools-config.cmake")
```

Comment:
I'll add a comment, sure, but what this is doing is generating a snippet of CMake code that creates an `IMPORTED` target inside the `hannk_tools-config.cmake` file that corresponds to `${TARGET}`. For an executable it would look something like:

```
add_exectuable(hannk::tools::${TARGET} IMPORTED)
set_target_properties(hannk::tools::${TARGET} PROPERTIES IMPORTED_LOCATION "/path/to/${TARGET}")
```

Then, when the cross build loads this file (via `find_package`), it can freely use `hannk::tools::${TARGET}`. The `APPEND` argument just allows us to build up the file incrementally.

Reply:
Added a comment.